zulip_bots/zulip_bots/test_lib.py

```diff
@@ -95,24 +96,52 @@ def check_expected_responses(self, expectations, expected_method='send_reply',
                 # For calls with send_message, r is already a Dict.
                 message = dict(trigger_message, content = m)
                 response = {'content': r} if expected_method == 'send_reply' else r
-                self.assert_bot_response(message=message, response=response,
-                                         expected_method=expected_method,
-                                         state_handler=current_state_handler)
+                self.assert_bot_responses(message, (response, expected_method),
+                                          state_handler=state_handler)
 
-    def call_request(self, message, expected_method, response, state_handler):
-        # type: (Dict[str, Any], str, Dict[str, Any], Optional[StateHandler]) -> None
+    def call_request(self, message, *responses, **kwargs):
+        # type: (Dict[str, Any], *Tuple[Dict[str, Any], str], **Any) -> None
+        state_handler = kwargs.pop('state_handler', None)
         if state_handler is None:
             state_handler = StateHandler()
-        # Send message to the concerned bot
-        self.message_handler.handle_message(message, self.MockClass(), state_handler)
 
-        # Check if the bot is sending a message via `send_message` function.
-        # Where response is a dictionary here.
+        # Mock BotHandler; get instance
+        bot_handler = self.MockClass()
         instance = self.MockClass.return_value
-        if expected_method == "send_message":
-            instance.send_message.assert_called_with(response)
-        else:
-            instance.send_reply.assert_called_with(message, response['content'])
+
+        # Send message to the bot
+        try:
+            self.message_handler.handle_message(message, bot_handler, state_handler)
+        except KeyError as key_err:
+            raise Exception("Message tested likely required key {}.".format(key_err))
+
+        # Determine which messaging functions are expected
+        send_messages = [call(r[0]) for r in responses if r[1] == 'send_message']
```

Comment:
I think `unittest.call` would make this statement clearer.

Reply:
Sorry, I don't understand what you mean.

Reply:
Explicitly writing `unittest.call` instead of `call` in this case. But I'm surprised this comment is still here? I thought I deleted it - it's not really important, after all.